test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlCommentsSchemaFilterTests.cs

```diff
@@ -75,8 +84,14 @@ public class XmlCommentsSchemaFilterTests
 
             Subject().Apply(schema, filterContext);
 
-            Assert.IsType<OpenApiString>(schema.Properties[propertyName].Example);
-            Assert.Equal(expectedExample, ((OpenApiString)schema.Properties[propertyName].Example).Value);
+            var openApiPrimitive = (IOpenApiPrimitive)schema.Properties[propertyName].Example;
+            Assert.Equal(expectedPrimitiveType, openApiPrimitive.PrimitiveType);
+            Assert.Equal(expectedValueString, GetOpenApiPrimitiveValue(openApiPrimitive).ToString());
+        }
```

Comment:
Thanks for the quick turnaround. One last change before I merge. Prior to the assertions, you can cast `Example` to the concrete `OpenApiPrimitive` instead of the interface. The concrete type has a `Value` property that you can use instead of the reflection hack. For example:

```csharp
var openApiPrimitive = (OpenApiPrimitive)schema.Properties[propertyName].Example;
Assert.Equal(expectedValueString, openApiPrimitive.Value.ToString());
```

Reply:
Thank you for the review!
I would like to cast to `OpenApiPrimitive`, but I can't: `OpenApiPrimitive` is generic type (https://github.com/Microsoft/OpenAPI.NET/blob/master/src/Microsoft.OpenApi/Any/OpenApiPrimitive.cs#L14). :-(
That's why I had to add this ugly reflection hack.